test/serializer/abstract/ToString3.js

```diff
@@ -0,0 +1,3 @@
+let x = global.__abstract ? __abstract("number", "(3)") : 3;
+let ob =  { valueOf: () => x };
+inspect = function() { return 123 * ob; }
```

Comment:
Since the * operation is inside a residual function, it will not be prepacked and hence the code for dealing with with abstract conversion values will not kick in. Add "let y = 123 * ob;" to the global code and then just reference y in the residual function.

Reply:
Is this because only global code (or code that runs during initialisation phase) get prepacked, as explained in the note below Module Initialization section on the landing page?

If thats right, these test files are nothing but small standalone programs that are getting prepacked themselves with certain comments a like acting like assertions? Is `inspect` analogous to `__residual`?

Reply:
All functions that can be reached from the external environment are residual. The global object is treated as externally visible.

If inspect were called in the global code itself, the call would be inlined (and thus prepacked), but the function will remain (because it is visible).